nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java

```diff
@@ -96,25 +101,52 @@ protected MongoClient createClient(ConfigurationContext context, MongoClient exi
             final String user = context.getProperty(DB_USER).evaluateAttributeExpressions().getValue();
             final String passw = context.getProperty(DB_PASSWORD).evaluateAttributeExpressions().getValue();
 
             final MongoClientSettings.Builder builder = MongoClientSettings.builder();
-            final ConnectionString cs = new ConnectionString(uri);
 
-            if (user != null && passw != null) {
+            if (StringUtils.containsIgnoreCase(uri, "authmechanism") && user != null && passw != null) {
+                // we extract all specified parameters in the URI
+                final Map<String, String> result = new LinkedHashMap<>();
+                for (String param : uri.split("&")) {
+                    String[] pair = param.split("=", 2);
+                    String key = URLDecoder.decode(pair[0], StandardCharsets.UTF_8).toLowerCase();
+                    String value = pair.length > 1 ? URLDecoder.decode(pair[1], StandardCharsets.UTF_8) : "";
+                    result.put(key, value);
+                }
+
+                uri = uri.replaceAll("(?i)authmechanism=[^&]*&?", "");
+                final ConnectionString cs = new ConnectionString(uri);
                 final String database = cs.getDatabase() == null ? "admin" : cs.getDatabase();
-                final MongoCredential credential = MongoCredential.createCredential(user, database, passw.toCharArray());
-                builder.credential(credential);
+                final AuthenticationMechanism mechanism = AuthenticationMechanism.fromMechanismName(result.get("authmechanism").toUpperCase());
+
+                switch (mechanism) {
+                    case SCRAM_SHA_1 -> builder.credential(MongoCredential.createScramSha1Credential(user, database, passw.toCharArray()));
+                    case SCRAM_SHA_256 -> builder.credential(MongoCredential.createScramSha256Credential(user, database, passw.toCharArray()));
+                    case MONGODB_AWS -> builder.credential(MongoCredential.createAwsCredential(user, passw.toCharArray()));
+                    case PLAIN -> builder.credential(MongoCredential.createPlainCredential(user, database, passw.toCharArray()));
+                    default -> throw new IllegalArgumentException("Unsupported authentication mechanism with username and password: " + mechanism);
+                }
+
+                builder.applyConnectionString(cs);
+            } else {
+                final ConnectionString cs = new ConnectionString(uri);
+
+                if (user != null && passw != null) {
+                    final String database = cs.getDatabase() == null ? "admin" : cs.getDatabase();
+                    final MongoCredential credential = MongoCredential.createCredential(user, database, passw.toCharArray());
+                    builder.credential(credential);
+                }
+
+                builder.applyConnectionString(cs);
```

Comment:
This call can be moved outside the conditional, using the `final` value of the `ConnectionString` object from either branch.